db/migrate/20200521150213_create_storage_services.rb

```diff
@@ -0,0 +1,18 @@
+class CreateStorageServices < ActiveRecord::Migration[5.1]
+  def change
+    create_table :storage_services do |t|
+      t.string :name
+      t.string :description
+      t.bigint :project_id
+      t.bigint :profile_id
+      t.string :uuid
+      t.bigint :version
+      t.references :parent_service, :type => :bigint, :index => true, :references => :storage_service
```

Comment:
Is there a limitation on how many parents a service can have?  We do have the ancestry gem which helps model tree-like structures in the database